### DISCUSSION TITLE
Can't get Bluetooth or Bluetooth LE to work in a C# Console App

### DISCUSSION THREAD

**Discussion by ghost** [inthehand/32feet#346] [Q&A] [answered]
When I tried the Bluetooth LE nuget I got an error "One or more errors occurred. (No path specified for UNIX transport)'"

I've tried both nuget packages with the sample console apps here in Github and I can get neither of them to work.

**Comment by peterfoot** [comment, marked as answer]
This happens when the generic net60 (or net70) dll is used from the NuGet rather than the Windows specific one. I'm looking into a way of improving this as it's not clear and currently the unix codebase doesn't throw PlatformNotSupported exception when called on another (probably Windows) platform. The reason for this is usually because the consuming project is a generic .NET project and doesn't specify a platform. It looks like the solution to this is to change the generic dll to include both Windows and Linux runtime support which will make the library bigger but simplify deployment.

**Reply by ghost** [reply]
Thank you for responding, I did switch back to .NET Framework and that allowed me to connect to one of the two printers I have.  We are dealing with some printers that look like Zebra's and we are told they support ZPL. Still, what we are finding they are some weird hybrid so I needed an easy way outside of our mobile application to connect easily to these printers and send some ZPL to them to help figure out the issues.

Once I figured out the Service UUID I figured out you can't just connect you have to Pair with it and then connect even though from Windows you can't pair with the printer it tells you that you need software but from code, if I "Search", "Pair", "Connect", "Send" that seemed to do the trick.

Once I switched back to .NET Framework, I could at least connect to one of them.  Figuring out the Service UUID took a while but it ended up being Serial and not one of the "Printer" UUID's.  The list of available services is blank but I'm betting that's the printer not returning them and not the library not getting them.

**Comment by longlong9568** [comment]
var devices = await Bluetooth.ScanForDevicesAsync();

  
  foreach (var device in devices)
  {
      Console.WriteLine($"Device: {device.Name}, Address: {device.Id}");
  } 
hi ,How do I handle this error when using Netcore 8.0? No path specified for UNIX transport
